flow/scripts/sc/flows/orflow.py

```diff
@@ -37,39 +37,77 @@ def setup(chip, flowname='orflow'):
-        'tapcell': ['openroad', 'tapcell.tcl'],
-        'pdn': ['openroad', 'pdn.tcl'],
-        'gp_skip_io': ['openroad', 'global_place_skip_io.tcl'],
-        'io_place': ['openroad', 'io_placement.tcl'],
-        'global_place': ['openroad', 'global_place.tcl'],
-        'resize': ['openroad', 'resize.tcl'],
-        'detail_place': ['openroad', 'detail_place.tcl'],
-        'clock_tree_syn': ['openroad', 'cts.tcl'],
-        'fillcells': ['openroad', 'fillcell.tcl'],
-        'global_route': ['openroad', 'global_route.tcl'],
-        'detail_route': ['openroad', 'detail_route.tcl'],
-        'final_report': ['openroad', 'final_report.tcl'],
-        # Like yosys, KLayout GDS-streaming script is run through an OpenROAD tcl script
-        'export': ['openroad', 'klayout.tcl'],
+
+    flow_groups = {
+        'synthesis': [
+            ('or_yosys', 'openroad'), # (synthesis is done via OpenROAD TCL that calls yosys)
+        ],
+        'floorplan': [
+            ('or_floorplan', 'openroad'),
+            ('or_io_placement_random', 'openroad'),
+            ('or_tdms_place', 'openroad'),
+            ('or_macro_place', 'openroad'),
+            ('or_tapcell', 'openroad'),
+            ('or_pdn', 'openroad'),
+        ],
+        'place': [
+            ('or_global_place_skip_io', 'openroad'),
+            ('or_io_placement', 'openroad'),
+            ('or_global_place', 'openroad'),
+            ('or_resize', 'openroad'),
+            ('or_detail_place', 'openroad'),
+        ],
+        'cts': [
+            ('or_cts', 'openroad'),
+            ('or_fillcell', 'openroad'),
+        ],
+        'route': [
+            ('or_global_route', 'openroad'),
+            ('or_detail_route', 'openroad'),
+        ],
+        'finish': [
+            ('or_final_report', 'openroad'),
+            # Like yosys, KLayout GDS-streaming script is run through an OpenROAD tcl script
+            ('export', 'openroad')
+        ]
+    }
+    flowpipe = ['synthesis', 'floorplan', 'place', 'cts', 'route', 'finish']
+
+    # Additional dependencies defined here
+    extra_deps = {
+        'or_detail_route': ['or_fillcell'], # reads 4_cts.odb
     }
-    last_step = ''
-    for k,v in flowpipe.items():
-        chip.node(flow, k, v[0])
-        if last_step:
-            chip.edge(flow, last_step, k)
-        if v[1]:
-            chip.set('tool', v[0], 'script', k, '0',
-                     os.path.abspath(os.path.join(openroad_dir, 'flow', 'scripts', v[1])))
-            chip.set('tool', v[0], 'refdir', k, '0', os.path.join('tools', v[0]))
-        last_step = k
+
+    chip.node(flow, 'import', 'nop')
+    last_step = 'import'
+    last_sdc_step = ''
+    for group in flowpipe:
+        for step, tool in flow_groups[group]:
+            chip.node(flow, step, tool)
+
+            if last_step:
+                chip.edge(flow, last_step, step)
+            if last_sdc_step and last_sdc_step != last_step:
+                chip.edge(flow, last_sdc_step, step)
```

Comment:
Will this dependency make the flow automatically copy (for example) `outputs/2_floorplan.sdc` -> `inputs/3_place.sdc`, or do we still need the manual `shutil.copy` calls in the build script?